modules/devkit-support/src/main/java/org/mule/security/oauth/processor/BaseOAuth2AuthorizeMessageProcessor.java

```diff
@@ -134,12 +136,16 @@ private void setState(Map<String, String> extraParameters, MuleEvent event)
 
                 try
                 {
+                    JsonProperty property=field.getAnnotation(JsonProperty.class);
```

Comment:
What you're proposing still doesn't seem like the best solution to me.

AuthorizationParameter is a DTO used to bridge the Devkit way of representing things (which are mainly annotations) against Mule's. Additionally, If you define a new annotation then it belongs on Devkit's codebase not mule's. If it's on devkit's, then mule cannot access it because there're dependency issues which prevents Mule from knowing devkit's jars.

I would simply add a getAlias() method to AuthorizationParameter. So, when the field is fetch you use the name, but when you build the extra parameters map which is used to build the request you use the alias. In this way the design remains consistent, in the sense that AuthorizationParameter remains a DTO and the two worlds remain separated.